src/PixelFlags.cc

```diff
@@ -57,87 +59,153 @@ class FootprintBits : public afw::detection::FootprintFunctor<MaskedImageT> {
-        {"flag_cr", "Cosmic ray in the source footprint"},
-        {"flag_crCenter", "Cosmic ray in the source center"},
-        {"flag_bad", "Bad pixel in the source footprint"}
-    }};
-    _flagHandler = FlagHandler::addFields(schema, name, flagDefs.begin(), flagDefs.end());
+    // Add generic keys first, which don't correspond to specific mask planes
+    _generalFailure = schema.addField<afw::table::Flag>(name+"_"+_ctrl.fname,
+                                        "general failure flag, set if anything went wring");
+    _offImageKey = schema.addField<afw::table::Flag>(name+"_"+_ctrl.fname+"_offimage", 
+                                        "source center is off image");
+    // Set all the flags that correspond to mask planes anywhere in the footprint
+    _predefinedAnyKeys["EDGE"] = schema.addField<afw::table::Flag>(name+"_"+_ctrl.fname+"_edge",
+                                        "Source is outside usable exposure region (masked EDGE or NO_DATA)");
+    _predefinedAnyKeys["INTRP"] = schema.addField<afw::table::Flag>(name+"_"+_ctrl.fname+"_interpolated",
+                                        "Interpolated pixel in the source footprint");
+    _predefinedAnyKeys["SAT"] = schema.addField<afw::table::Flag>(name+"_"+_ctrl.fname+"_saturated",
+                                        "Saturated pixel in the source footprint");
+    _predefinedAnyKeys["CR"] = schema.addField<afw::table::Flag>(name+"_"+_ctrl.fname+"_cr",
+                                        "Cosmic ray in the source footprint");
+    _predefinedAnyKeys["BAD"] = schema.addField<afw::table::Flag>(name+"_"+_ctrl.fname+"_bad",
+                                        "Bad pixel in the source footprint");
+    _predefinedAnyKeys["SUSPECT"] = schema.addField<afw::table::Flag>(name+"_"+_ctrl.fname+"_suspect",
+                                        "source's footprint includes suspect pixels");
+    // Flags that correspond to mask bits which occur in the center of the object
+    _predefinedCenterKeys["INTRP"] = schema.addField<afw::table::Flag>(name+"_"+_ctrl.fname+"_interpolatedCenter",
+                                        "Interpolated pixel in the source center");
+    _predefinedCenterKeys["SAT"] = schema.addField<afw::table::Flag>(name+"_"+_ctrl.fname+"_saturatedCenter",
+                                        "Saturated pixel in the source center");
+    _predefinedCenterKeys["CR"] = schema.addField<afw::table::Flag>(name+"_"+_ctrl.fname+"_crCenter",
+                                        "Cosmic ray in the source center");
+    _predefinedCenterKeys["SUSPECT"] = schema.addField<afw::table::Flag>(name+"_"+_ctrl.fname+"_suspectCenter",
+                                        "source's center is close to suspect pixels");
+
+    // Read in the flags passed from the configuration, and add them to the schema
+    for (std::vector<std::string>::const_iterator i = _ctrl.center.begin(); i != _ctrl.center.end(); ++i){
+        std::string tname(*i);
```

Comment:
I realise you're using `tname` to mean "transformed name" here. But you already have a `name` variable... which is not the one you're transforming. Can you find something more descriptive?